### PR TITLE
Rebuild all packages that have an incorrect file size in the sync databases

### PR DESCRIPTION
**Pull request by cbrt64**
I'm relatively new to contributing, please let me know if there's a better way to go about this.

Exhaustively comparing the (presumably correct) file sizes from [the repo listings](http://repo.msys2.org/msys/) to the descriptions in both msys.db files shows ten total packages with bogus file sizes listed in msys.db. This is the root cause of #942 and friends. A simple rebuild fixed this for ucl; this PR is intended to fix the remainder.

Packages with incorrect sizes in db | [i686](http://repo.msys2.org/msys/x86_64/msys.db) | [x86_64](http://repo.msys2.org/msys/x86_64/msys.db)
-- | -- | --
ctags, gengetopt, perl-Compress-Bzip2, txt2tags | X | X
idutils, task | | X

This also agrees with @lockywolf's list in #1025.

### CODE REVIEW

**Comment by cbrt64** [comment]
TL;DR Looks like a one-time update mix-up over a year ago. @Alexpux builds the packages, he might be able to confirm?

I wondered about this myself, so I compared the [repos](http://repo.msys2.org/msys/) to some older `msys.db` files:

Before this PR, the affected packages each had only one version uploaded, except `perl-Compress-Bzip2`, which has older versions on Sourceforge. For that one, the incorrect package sizes in `msys.db` were more consistent (though not exact) with these older versions. Also, the package build dates in `msys.db` were much older than the packages themselves. This implies that at one point the build machine had some packages that were added to the repo db, but not uploaded; then later those (same version) packages were rebuilt and uploaded but _not_ added to the db.

All of the affected packages were part of a large batch of 160 packages uploaded between 2017-05-31 and 2017-06-02, and none of them had been updated since. I suspect at that time a number of new packages missed being `repo-add`ed.

This problem is not limited to MSYS2; there are similar reports for AUR and Antergos, for example, that end up magically fixing themselves when the packages are updated again.

The ultimate problem as I see it is this: If Arch/pacman has no documentation for how to automate the maintaining of a repo (I can't find it anyway), then anyone running a repo has to roll their own. And if it's not automated or scripted, the maintainer might sometimes miss a step in the process of doing `makepkg && repo-add && rsync` or whatever.

**Comment by earnie** [comment]
On 11/5/2018 8:29 PM, Edward E. wrote:
> TL;DR Looks like a one-time update mix-up over a year ago. @Alexpux
> <https://github.com/Alexpux> builds the packages, he might be able to
> confirm?
> 

Thanks Edward, I assume the same thing.

> I wondered about this myself, so I compared the repos
> <http://repo.msys2.org/msys/> to some older |msys.db| files:
> 
> Before this PR, the affected packages each had only one version
> uploaded, except |perl-Compress-Bzip2|, which has older versions on
> Sourceforge. For that one, the incorrect package sizes in |msys.db| were
> more consistent (though not exact) with these older versions. Also, the
> package build dates in |msys.db| were much older than the packages
> themselves. This implies that at one point the build machine had some
> packages that were added to the repo db, but not uploaded; then later
> those (same version) packages were rebuilt and uploaded but /not/ added
> to the db.
> 
> All of the affected packages were part of a large batch of 160 packages
> uploaded between 2017-05-31 and 2017-06-02, and none of them had been
> updated since. I suspect at that time a number of new packages missed
> being |repo-add|ed.
> 
> This problem is not limited to MSYS2; there are similar reports for AUR
> and Antergos, for example, that end up magically fixing themselves when
> the packages are updated again.
> 

I can see this as well.

> The ultimate problem as I see it is this: If Arch/pacman has no
> documentation for how to automate the maintaining of a repo (I can't
> find it anyway), then anyone running a repo has to roll their own. And
> if it's not automated or scripted, the maintainer might sometimes miss a
> step in the process of doing |makepkg && repo-add && rsync| or whatever.
> 

There is also a timing issue where the package files can be uploaded
prior to the DB being uploaded and some user doing a request for the
package.  Fortunately the package release (pkgrel) should be updated but
even this can be a failure point should someone forget to update it and
end up loading a package with the same exact name.

The bigger question is what has been put into place to help manage this
so that it doesn't occur again?

-- 
Earnie

**Comment by lygstate** [comment]
```
downloading mingw-w64-x86_64-mpdecimal-2.5.0-1-any.pkg.tar.zst...
downloading mingw-w64-x86_64-libiconv-1.16-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-gettext-0.19.8.1-9-any.pkg.tar.zst...
error: failed retrieving file 'mingw-w64-x86_64-libtre-git-r128.6fb7206-2-any.pkg.tar.xz' from sourceforge.net : Maximum file size exceeded
downloading mingw-w64-x86_64-libtre-git-r128.6fb7206-2-any.pkg.tar.xz...
error: failed retrieving file 'mingw-w64-x86_64-libsystre-1.0.1-4-any.pkg.tar.xz' from sourceforge.net : Maximum file size exceeded
downloading mingw-w64-x86_64-libsystre-1.0.1-4-any.pkg.tar.xz...
downloading mingw-w64-x86_64-ncurses-6.2-2-any.pkg.tar.zst...
downloading mingw-w64-x86_64-libtasn1-4.16.0-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-p11-kit-0.23.20-2-any.pkg.tar.xz...
downloading mingw-w64-x86_64-ca-certificates-20200601-1-any.pkg.tar.zst...
downloading mingw-w64-x86_64-zlib-1.2.11-7-any.pkg.tar.xz...
downloading mingw-w64-x86_64-openssl-1.1.1.h-1-any.pkg.tar.zst...
downloading mingw-w64-x86_64-termcap-1.3.1-6-any.pkg.tar.zst...
downloading mingw-w64-x86_64-readline-8.0.004-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-tcl-8.6.10-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-sqlite3-3.33.0-1-any.pkg.tar.zst...
downloading mingw-w64-x86_64-tk-8.6.10-2-any.pkg.tar.zst...
downloading mingw-w64-x86_64-xz-5.2.5-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-python-3.8.6-3-any.pkg.tar.zst...
downloading mingw-w64-x86_64-python-pyparsing-2.4.7-1-any.pkg.tar.zst...
downloading mingw-w64-x86_64-python-six-1.15.0-1-any.pkg.tar.zst...
downloading mingw-w64-x86_64-python-attrs-19.3.0-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-python-packaging-20.3-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-python-ordered-set-3.1.1-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-python-appdirs-1.4.3-1-any.pkg.tar.xz...
downloading mingw-w64-x86_64-python-setuptools-47.1.1-1-any.pkg.tar.zst...
downloading mingw-w64-x86_64-binutils-2.35.1-2-any.pkg.tar.zst...
downloading mingw-w64-x86_64-headers-git-8.0.0.6001.98dad1fe-1-any.pkg.tar.zst...
downloading mingw-w64-x86_64-crt-git-8.0.0.6001.98dad1fe-1-any.pkg.tar.zst...
downloading mingw-w64-x86_64-isl-0.22.1-1-any.pkg.tar.xz...
error: failed retrieving file 'mingw-w64-x86_64-windows-default-manifest-6.4-3-any.pkg.tar.xz' from sourceforge.net : Maximum file size exceeded
downloading mingw-w64-x86_64-windows-default-manifest-6.4-3-any.pkg.tar.xz...
downloading mingw-w64-x86_64-winpthreads-git-8.0.0.6001.98dad1fe-3-any.pkg.tar.zst...
downloading mingw-w64-x86_64-zstd-1.4.5-1-any.pkg.tar.zst...
```

**Comment by cbrt64** [comment]
@lygstate It seems those three files are currently missing from the sourceforge repo, so of course pacman is saying the server file sizes are too small, and falling back to your next configured repo.
